### PR TITLE
adding multi filter to Topic and Entity columns

### PR DESCRIPTION
**Pull request by sweng66**
1. adding a generic multi filter that will shows up as a typeahead if there are more than 10 options; otherwise shows up as a list of checkboxes for the options.
2. creating TopicFilter and EntityFilter by calling the generic MultiFilter
3. applying the TopicFilter and EntityFilter  to the Topic and Entity Columns in the Topic Entity Table
4. Renaming original EntityFilter to EntityTypeFilter (it is for entity types, not for entities)

### CODE REVIEW

File: src/components/AgGrid/EntityFilter.jsx [hunk 1]
```diff
@@ -1,15 +1,15 @@
 import { useGridFilter } from 'ag-grid-react';
 import React, { useCallback, useEffect, useState } from 'react';
-import {useSelector} from "react-redux";
-
+import { useSelector } from 'react-redux';
+import Select from 'react-select';
```
Comment (azurebrd) [inline src/components/AgGrid/EntityFilter.jsx:4]: Is react-select already in the repo, or do you need that in the package.json ?  If it's new, I'll defer to Valerio and Paul about whether we need that, or should use something else (I have no idea)

Comment (sweng66) [reply src/components/AgGrid/EntityFilter.jsx:4]: It is already in package.json.

Comment (sweng66) [reply src/components/AgGrid/EntityFilter.jsx:4]: "react-select": "^5.8.0"

**Review by azurebrd — COMMENTED** [review]
I still haven't worked with AgGrid, so I'll defer to other people.  In general looks good, just a question about whether we need react-select as an import in the package.json  and I'll defer to other people about that too.

File: src/components/AgGrid/EntityFilter.jsx (hunk 1 above)
Comment (azurebrd) [reply src/components/AgGrid/EntityFilter.jsx:4]: Ah, cool, thanks @sweng66 .  I didn't see it when I pulled main.

**Comment by sweng66** [comment]
Thanks! @azurebrd Good question.

File: src/components/AgGrid/EntityFilter.jsx (hunk 1 above)
Comment (azurebrd) [reply src/components/AgGrid/EntityFilter.jsx:4]: https://github.com/alliance-genome/agr_literature_ui/blob/main/package.json

Comment (sweng66) [reply src/components/AgGrid/EntityFilter.jsx:4]: You are right. For some reason, it is in my local version. I added the following to package.json.

"react-select": "^5.8.0"

Good catch! Thanks. @azurebrd

**Review by valearna — COMMENTED** [review]
These changes look good to me. I tried using the UI and it looks great. The component is a little small. Can you make it taller so that there's no double scrolling in the select @sweng66 ?

**Comment by sweng66** [comment]
@valearna I have fixed code to make the typeahead box taller.

**Comment by sweng66** [comment]
@valearna I has fixed the code. Can you check again in dev3006? Thanks!

**Review by valearna — APPROVED** [review]
Thanks @sweng66 !

**Pull request merged**